packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolAggregateEntities.ts

```diff
@@ -63,24 +61,11 @@ export const useBlockProtocolAggregateEntities = (): {
         };
       }
 
-      const { getAllLatestPersistedEntities: subgraph } = response;
-
-      /**
-       * @todo: remove this when we start returning links in the subgraph
-       *   https://app.asana.com/0/0/1203214689883095/f
-       */
-      for (const [_, vertex] of Object.entries(subgraph.vertices)) {
-        if (vertex.kind === "entity") {
-          (vertex.inner as unknown as Entity).links = [];
-        }
-      }
-
-      /**
-       * @todo: remove this when we start returning links in the subgraph
-       *   https://app.asana.com/0/0/1203214689883095/f
-       */
       return {
-        data: subgraph as Subgraph,
+        /** @todo - Is there a way we can ergonomically encode this in the GraphQL type? */
+        data: response.getAllLatestEntitiesWithMetadata as Subgraph<
+          SubgraphRootTypes["entity"]
+        >,
```

Comment:
We could create some `EntityRootedSubgraph` scalar?

Reply:
Yeah I think we should consider this, it might be worth it